### PR TITLE
everywhere: replace double words

### PR DESCRIPTION
**Pull request by LingaoM**
``` python
import os
import re

common_words = set([
    'about', 'after', 'all', 'also', 'an', 'and',
     'any', 'are', 'as', 'at',
    'be', 'because', 'but', 'by', 'can', 'come',
    'could', 'day', 'do', 'even',
    'first', 'for', 'get', 'give', 'go', 'has',
    'have', 'he', 'her',
    'him', 'his', 'how', 'I', 'in', 'into', 'it',
    'its', 'just',
    'know', 'like', 'look', 'make', 'man', 'many',
    'me', 'more', 'my', 'new',
    'no', 'not', 'now', 'of', 'one', 'only', 'or',
    'other', 'our', 'out',
    'over', 'people', 'say', 'see', 'she', 'so',
    'some', 'take', 'tell', 'than',
    'their', 'them', 'then', 'there', 'these',
    'they', 'think',
    'this', 'time', 'two', 'up', 'use', 'very',
    'want', 'was', 'way',
    'we', 'well', 'what', 'when', 'which', 'who',
    'will', 'with', 'would',
    'year', 'you', 'your'
])

valid_extensions = set([
    'c', 'h', 'yaml', 'cmake', 'conf', 'txt', 'overlay', 'rst', 'dtsi',
    'Kconfig', 'dts', 'defconfig', 'yml', 'ld', 'sh', 'py', 'soc', 'cfg'
])

def filter_repeated_words(text):
    # Split the text into lines
    lines = text.split('\n')

    # Combine lines into a single string with unique separator
    combined_text = '/*sep*/'.join(lines)

    # Replace repeated words within a line
    def replace_within_line(match):
        return match.group(1)

    # Regex for matching repeated words within a line
    within_line_pattern = re.compile(r'\b(' + '|'.join(map(re.escape, common_words)) + r')\b\s+\b\1\b')
    combined_text = within_line_pattern.sub(replace_within_line, combined_text)

    # Replace repeated words across line boundaries
    def replace_across_lines(match):
        return match.group(1) + match.group(2)

    # Regex for matching repeated words across line boundaries
    across_lines_pattern = re.compile(r'\b(' + '|'.join(map(re.escape, common_words)) + r')\b(\s*[*\/\n\s]*)\b\1\b')
    combined_text = across_lines_pattern.sub(replace_across_lines, combined_text)

    # Split the text back into lines
    filtered_text = combined_text.split('/*sep*/')

    return '\n'.join(filtered_text)

def process_file(file_path):
    with open(file_path, 'r', encoding='utf-8') as file:
        text = file.read()

    new_text = filter_repeated_words(text)

    with open(file_path, 'w', encoding='utf-8') as file:
        file.write(new_text)

def process_directory(directory_path):
    for root, dirs, files in os.walk(directory_path):
        dirs[:] = [d for d in dirs if not d.startswith('.')]
        for file in files:
            # Filter out hidden files
            if file.startswith('.'):
                continue
            file_extension = file.split('.')[-1]
            if file_extension in valid_extensions:  # 只处理指定后缀的文件
                file_path = os.path.join(root, file)
                print(f"Processed file: {file_path}")
                process_file(file_path)

directory_to_process = "/home/mi/works/github/zephyrproject/zephyr"
process_directory(directory_to_process)
```

### CODE REVIEW

**Comment by LingaoM** [comment]
As a supplement to this PR:https://github.com/zephyrproject-rtos/zephyr/pull/74650

File: arch/xtensa/core/README_MMU.txt [hunk 1]
```diff
@@ -254,7 +254,7 @@ of access variability.  But it also means that the TLB entries end up
 being stored twice in the same CPU, wasting transistors that could
 presumably store other useful data.
 
-But it it also important to note that the L1 data cache on Xtensa is
+But it also important to note that the L1 data cache on Xtensa is
```
Comment (kartben) [inline arch/xtensa/core/README_MMU.txt:257]:
```suggestion
But it is also important to note that the L1 data cache on Xtensa is
```

File: doc/connectivity/networking/api/lwm2m.rst [hunk 2]
```diff
@@ -670,7 +670,7 @@ where it cannot recover.
      - Try to recover network connection. Then restart the client by calling :c:func:`lwm2m_rd_client_start`.
        This might also indicate configuration issue.
 
-Sending of data in the table above refers to calling :c:func:`lwm2m_send_cb` or by writing into of of the observed resources where observation would trigger a notify message.
+Sending of data in the table above refers to calling :c:func:`lwm2m_send_cb` or by writing into of the observed resources where observation would trigger a notify message.
```
Comment (kartben) [inline doc/connectivity/networking/api/lwm2m.rst:673]:
```suggestion
Sending of data in the table above refers to calling :c:func:`lwm2m_send_cb` or by writing into one of the observed resources where observation would trigger a notify message.
```

File: samples/subsys/usb/webusb/src/main.c [hunk 3]
```diff
@@ -64,7 +64,7 @@ static struct msosv2_descriptor_t {
 		.wLength = sizeof(struct msosv2_function_subset_header),
 		.wDescriptorType = MS_OS_20_SUBSET_HEADER_FUNCTION,
 		/* The WebUSB interface number becomes the first when CDC_ACM is enabled by
-		 * configuration.  Beware that if this sample is used as as inspiration for
+		 * configuration.  Beware that if this sample is used as inspiration for
```
Comment (kartben) [inline samples/subsys/usb/webusb/src/main.c:67]:
```suggestion
		 * configuration.  Beware that if this sample is used as an inspiration for
```

File: subsys/mgmt/mcumgr/grp/img_mgmt/src/zephyr_img_mgmt.c [hunk 4]
```diff
@@ -186,7 +186,7 @@ img_mgmt_flash_area_id(int slot)
  * find any unused and non-active available (auto-select); any other positive
  * value is direct (slot + 1) to be used; if checks are positive, then area
  * ID is returned, -1 is returned otherwise.
- * Note that auto-selection is performed only between two two first slots.
+ * Note that auto-selection is performed only between two first slots.
```
Comment (kartben) [inline subsys/mgmt/mcumgr/grp/img_mgmt/src/zephyr_img_mgmt.c:189]:
```suggestion
 * Note that auto-selection is performed only between the two first slots.
```

**Review by kartben — CHANGES_REQUESTED** [review]

File: arch/arc/core/mpu/arc_mpu_v2_internal.h [hunk 5]
```diff
@@ -118,7 +118,7 @@ static inline bool _is_enabled_region(uint32_t r_index)
 }
 
 /**
- * This internal function check if the given buffer in in the region
+ * This internal function check if the given buffer in the region
```
Comment (kartben) [inline arch/arc/core/mpu/arc_mpu_v2_internal.h:121]:
```suggestion
 * This internal function check if the given buffer is in the region
```

File: arch/arc/core/mpu/arc_mpu_v6_internal.h [hunk 6]
```diff
@@ -156,7 +156,7 @@ static inline bool _is_enabled_region(uint32_t r_index)
 }
 
 /**
- * This internal function check if the given buffer in in the region
+ * This internal function check if the given buffer in the region
```
Comment (kartben) [inline arch/arc/core/mpu/arc_mpu_v6_internal.h:159]:
```suggestion
 * This internal function check if the given buffer is in the region
```

**Review by kartben — CHANGES_REQUESTED** [review]
Spotted more, sorry

**Review by jhedberg — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Comment by LingaoM** [comment]
@kartben Please take a look, tks.

**Review by kartben — APPROVED** [review]

**Pull request merged**